Source/Schema.NET/ValuesJsonConverter.cs

```diff
@@ -222,296 +163,190 @@ public virtual void WriteObject(JsonWriter writer, object value, JsonSerializer
                     }
-                }
-                else
-                {
-                    args = Convert.ToDecimal(value, CultureInfo.InvariantCulture);
-                }
-            }
-            else if (unwrappedType == typeof(DateTime))
-            {
-                if (value is string)
-                {
-                    if (DateTime.TryParse((string)value, CultureInfo.InvariantCulture, DateTimeStyles.None, out var i))
+                    else
                     {
-                        args = i;
+                        enumString = valueString;
                     }
-                }
-                else if (value is DateTime)
-                {
-                    args = value;
-                }
-                else if (value is DateTimeOffset)
-                {
-                    args = ((DateTimeOffset)value).DateTime;
-                }
-                else if (value is short || value is int || value is long || value is float || value is double)
-                {
-                    // NO-OP: can't put a primitive type into a date
-                }
-                else
-                {
-                    args = Convert.ToDateTime(value, CultureInfo.InvariantCulture);
-                }
-            }
-            else if (unwrappedType == typeof(DateTimeOffset))
-            {
-                if (value is string)
-                {
-                    if (DateTimeOffset.TryParse((string)value, CultureInfo.InvariantCulture, DateTimeStyles.None, out var i))
+
+                    try
                     {
-                        args = i;
+                        result = Enum.Parse(targetType, enumString);
+                        success = true;
                     }
+#pragma warning disable CA1031 // Do not catch general exception types
+                    catch (Exception ex)
+                    {
+                        Debug.WriteLine(ex.Message);
+                        success = false;
+                    }
+#pragma warning restore CA1031 // Do not catch general exception types
                 }
-                else if (value is DateTime)
+                else if (targetType == typeof(DateTime))
                 {
-                    args = new DateTimeOffset((DateTime)value);
+                    success = DateTime.TryParse(valueString, CultureInfo.InvariantCulture, DateTimeStyles.None, out var localResult);
+                    result = localResult;
                 }
-                else if (value is DateTimeOffset)
+                else if (targetType == typeof(DateTimeOffset))
                 {
-                    args = value;
+                    success = DateTimeOffset.TryParse(valueString, CultureInfo.InvariantCulture, DateTimeStyles.None, out var localResult);
+                    result = localResult;
                 }
-                else
+                else if (targetType == typeof(Uri))

```

Comment:
I wonder if `Guid` should also be here, if we are checking a bunch of types? Not that schema.org has any Guid types (that I know of) but people might want to create custom types with Guids.

Reply:
Yeah, GUID might be a good one to add and wouldn't be hard to do.

Reply:
Added in d4f08f57 including test.